WordPress/gutenberg
# Migrate from TextControl to NumberControl to remove margin overrides

## What?



Updated `TextControl` instances using type number to `NumberControl` component. 

## Why?


While working on [this project](https://github.com/WordPress/gutenberg/issues/38730) to remove margin overrides, I thought it made more sense to use NumberControl which also helped with removing overrides. 

## How?


By adding NumberControl, HStack, and VStack and removing unnecessary CSS. 

## Testing Instructions


- ### ImageSizeControl

	1. Add Image block and upload image
	2. Click on the settings icon in block inspector
	3. Look for 'Width' and 'Height' labels
	4. Verify spacing and that component works the same as before
	5. Also, check visuals in Safari to make sure it looks the same as other browsers (CSS was removed due to Safari-only issue that has since been resolved)
	
	| <br /> Before  | <br />After |
	| --- | --- |
	| <img width="280" alt="Screenshot 2023-01-13 at 11 52 09 PM" src="https://user-images.githubusercontent.com/35543432/212463128-0b70adc9-df68-4acf-8db1-ab017ca18a4a.png"> | <img width="285" alt="Screenshot 2023-01-13 at 11 52 17 PM" src="https://user-images.githubusercontent.com/35543432/212463129-98f97586-5787-4d6e-97ed-e4ec86b75715.png"> |

- ### InlineUI

	1. Add a paragraph block with text
	2. Click on the arrow in the toolbar and select 'Inline Image'
	3. Add an image and once it's in the paragraph, select it
	5. Ensure the space below the text input 'Width' is the same as before 
	6. Verify component works the same as before 
	
	| <br /> Before  | <br />After |
	| --- | --- |
	| <img width="221" alt="Screenshot 2023-01-13 at 9 12 40 PM" src="https://user-images.githubusercontent.com/35543432/212463087-b9418a1f-3893-4a6c-8b06-ea8113102a58.png"> | <img width="222" alt="Screenshot 2023-01-13 at 9 12 35 PM" src="https://user-images.githubusercontent.com/35543432/212463084-89749924-a212-462b-bef8-269872926968.png"> |

- ### PageAttributesOrder

	1. Edit a page
	2. Click on 'Page Attributes'
	3. Ensure the space below the text input for 'Order' is the same as before 
	4. Verify component works the same as before
	 
	| <br /> Before  | <br />After |
	| --- | --- |
	| <img width="284" alt="Screenshot 2023-01-10 at 1 43 56 PM" src="https://user-images.githubusercontent.com/35543432/212463139-f608905d-1808-49dc-8f70-9a9c735f8449.png"> | <img width="288" alt="Screenshot 2023-01-10 at 1 44 03 PM" src="https://user-images.githubusercontent.com/35543432/212463236-61445dbc-be47-44b7-94d1-edb09f90f179.png"> |
